frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -0,0 +1,298 @@
+import * as React from 'react';
+import {
+  Alert,
+  Checkbox,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateVariant,
+  Select,
+  SelectOption,
+  Title,
+} from '@patternfly/react-core';
+import classnames from 'classnames';
+import { Trans, useTranslation } from 'react-i18next';
+import { coFetch } from '@console/internal/co-fetch';
+import {
+  ExternalLink,
+  getQueryArgument,
+  LoadingBox,
+  LoadingInline,
+  removeQueryArgument,
+  setQueryArgument,
+} from '@console/internal/components/utils';
+import { modelFor, NodeKind, resourceURL } from '@console/internal/module/k8s';
+import { useUserSettings } from '@console/shared';
+import { USERSETTINGS_PREFIX } from '@console/shared/src/constants';
+import NodeLogsFilterUnit from './NodeLogsUnitFilter';
+import './node-logs.scss';
+
+type NodeLogsProps = {
+  obj: NodeKind;
+};
+
+const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
+  const {
+    kind,
+    metadata: { labels, name, namespace: ns },
+    status,
+  } = node;
+  const isWindows = status?.nodeInfo.operatingSystem === 'windows';
```

Comment:
`nodeInfo` seems to be an optional property, so I'd add another `?`

```suggestion
  const isWindows = status?.nodeInfo?.operatingSystem === 'windows';
```